middlewares/optional_query_param.ts

```diff
@@ -20,21 +21,23 @@ import { fromEither, taskEither } from "fp-ts/lib/TaskEither";
 export const OptionalQueryParamMiddleware = <S, A>(
   name: string,
   type: t.Type<A, S>
-): IRequestMiddleware<"IResponseErrorValidation", Option<A>> => async (
-  request
-): Promise<Either<IResponse<"IResponseErrorValidation">, Option<A>>> =>
-  taskEither
-    .of<IResponse<"IResponseErrorValidation">, Option<unknown>>(
-      fromNullable(request.query[name])
-    )
-    .chain(maybeQuery =>
-      maybeQuery.foldL(
-        () => taskEither.of(none),
-        query =>
-          fromEither(type.decode(query)).bimap(
-            ResponseErrorFromValidationErrors(type),
-            some
+) => (
+  request: Request
+): Promise<E.Either<IResponseErrorValidation, O.Option<A>>> =>
+  pipe(
+    TE.of<IResponseErrorValidation, O.Option<unknown>>(
```

Comment:
ditto

Reply:
fixed in https://github.com/pagopa/io-functions-cgn-operator-search/pull/28/commits/8386c96583ac277673207742ac223f1fe84047fc